Yoast/wordpress-seo
# Run indexation JS on our tools page only

## Context


* To prevent JavaScript failures of others to break our indexation we want to run the indexation from our own page.

## Summary


This PR can be summarized in the following changelog entry:

* Moves the running of the SEO data indexing process to the Yoast Tools page.

## Relevant technical choices:

* After asking Herre, the conditional is loaded through DI instead of creating a new class and switching the conditionals around. A class gives more overhead that is not worth it for one variable changing.

## Test instructions

This PR can be tested by following these steps:

There are a couple of things to know before starting:
* You have to have more than 25 indexables that need to be indexed (you can truncate the indexable and indexable_hierarchy tables).
* There are some options we use to optionally hide the notification, you can use this code to make it appear again (thanks Andy):
```php
add_action( 'init', function() { 
    WPSEO_Options::set( 'ignore_indexation_warning', false );
    WPSEO_Options::set( 'indexation_warning_hide_until', false );
    WPSEO_Options::set( 'indexation_started', false );
} );
```

Tests:
* Visit the following pages to verify the start button is now a link:
  * All the Yoast admin pages, except the Tools page.
  * WP dashboard
  * WP plugins
  * WP update
  * WP permalinks settings
* The JS file `wp-seo-indexation-143-RC1.js?ver=14.3-RC1` should not be loaded on those pages. Note the version might've changed now.
* All those should link to the tools page and start the indexation.
* Visit the Yoast tools page and verify the start button is an actual button.
* Clicking it should start the indexation.

### Stop
* Start the indexation from another page than the Tools page.
* Stop the indexation.
* The hash in the URL should be removed and not start the indexation right away again 😅 

### Bonus: test the prominent words indexation on premium
* Merge this PR into the feature branch on premium.
* Visit a page other than the tools page and verify the indexation starts correctly (not starting the prominent words one).
* Check that the prominent words (internal linking) indexation still functions as before.

## UI changes

## Documentation

## Quality assurance


Fixes #15438
